frontend/src/components/explorer/mapViewer/ThreeDviewer.vue

```diff
@@ -52,30 +91,137 @@ export default {
       };
       await this.$store.dispatch('maps/get3DMapNetwork', payload);
       this.renderNetwork();
+      this.showLoader = false;
       // controller.filterBy({group: 'm'});
       // controller.filterBy({id: [1, 2, 3, 4]});
       // Subscribe to node selection events
       // document.getElementById('viewer').addEventListener('select', e => console.debug('selected', e.detail));
     },
-    renderNetwork() {
+    getElementIdAndType(element) {
+      if (element.group === 'r') {
+        return [element.id, 'reaction'];
+      } if (element.group === 'e') {
+        return [element.id.split('-')[0], 'gene'];
+      }
+      return [element.id.split('-')[0], 'metabolite'];
```

Comment:
I think the introduction of the compartmentalized metabolite has cleared the need for merging the metabolite id with `-neo4jid`. And I think we wouldn't need that for the genes. So I would expect to not have to use `split` over there.

Reply:
Good catch!